lib/src/view/tournament/tournament_screen.dart

```diff
@@ -0,0 +1,489 @@
+      screen: TournamentScreen(id: id),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    ref.listen(tournamentControllerProvider(id).select((value) => value.valueOrNull?.currentGame), (
+      prevGameId,
+      currentGameId,
+    ) {
+      if (prevGameId != currentGameId && currentGameId != null) {
+        Navigator.of(
+          context,
+          rootNavigator: true,
+        ).push(GameScreen.buildRoute(context, initialGameId: currentGameId));
+      }
+    });
+
+    return switch (ref.watch(tournamentControllerProvider(id))) {
+      AsyncError(:final error) => Center(child: Text('Could not load tournament: $error')),
+      AsyncValue(:final value?) => _Body(state: value),
+      _ => const PlatformScaffold(
+        appBarTitle: SizedBox.shrink(),
+        body: Center(child: CircularProgressIndicator()),
+      ),
+    };
+  }
+}
+
+class _Body extends ConsumerWidget {
+  const _Body({required this.state});
+
+  final TournamentState state;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return PlatformScaffold(
+      appBarTitle: _Title(state: state),
+      body: Column(
+        children: [
+          Expanded(
+            child: Padding(
+              padding: Styles.bodySectionPadding,
+              child: SingleChildScrollView(
+                child: Column(
+                  spacing: 20,
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    _Verdicts(state.tournament.verdicts),
+                    if (!state.tournament.berserkable)
+                      const Text.rich(
+                        TextSpan(
+                          children: [
+                            WidgetSpan(child: Icon(LichessIcons.body_cut, size: 16)),
+                            TextSpan(
+                              // TODO l10n
+                              text: ' No Berserk allowed',
+                            ),
+                          ],
+                        ),
+                      ),
+                    _Standing(state),
+                    if (state.tournament.featuredGame != null)
+                      _FeaturedGame(state.tournament.featuredGame!),
+                  ],
+                ),
+              ),
+            ),
+          ),
+          _BottomBar(state),
+        ],
+      ),
+    );
+  }
+}
+
+class _Title extends StatelessWidget {
+  const _Title({required this.state});
+
+  final TournamentState state;
+
+  @override
+  Widget build(BuildContext context) {
+    final timeLeft = state.tournament.timeToStart ?? state.tournament.timeToFinish;
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+      children: [
+        Flexible(child: AutoSizeText(state.tournament.fullName, maxLines: 1, minFontSize: 14.0)),
+        if (timeLeft != null)
+          CountdownClockBuilder(
+            timeLeft: timeLeft,
+            clockUpdatedAt: DateTime.now(),
```

Comment:
It doesn't make sense to pass `DateTime.now()` here. The date time should be part of the state and change each time we receive a new `timeLeft` value from the server.

But here we don't really care to compensate for UI lag, so you could also revert this commit in the PR:
https://github.com/lichess-org/mobile/commit/a8e71c0aac625f8d2c42859538c114a8d9adfed4